core/src/main/java/com/graphhopper/search/EdgeKVStorage.java

```diff
@@ -36,20 +39,43 @@ public class EdgeKVStorage {
     static final int MAX_UNIQUE_KEYS = (1 << 15);
     // Store string value as byte array and store the length into 1 byte
     private static final int MAX_LENGTH = (1 << 8) - 1;
+
+    // It stores the mapping of "key to index" in the keys DataAccess. E.g. if your first key is "some" then we will
+    // store the mapping "1->some" there (the 0th index is skipped on purpose). As this map is 'small' the keys
+    // DataAccess is only used for long term storage, i.e. only in loadExisting and flush. Otherwise we use keysInMem and keyList.
```

Comment:
But keys stores the "index to key" mapping right? I mean what we want is "key to index", but what we actually store is "index to key". 

keysInMem and keyList? You probably mean `Map<String, Integer> keyToIndex`, i.e. we read the keys DataAccess and keep the reverse mapping ("some"->1) in memory?